### PR TITLE
Better native firebase transcription

### PR DESCRIPTION
**Pull request by fabien0102**
### Checklist
- Issue number for this PR: #201
- Docs included?: yes
- Test units included?: yes
- e2e tests included?: no
- In a clean directory, `npm install`, `npm run build`, and `npm test` run successfully? yes
### Description

When I use angularfire2, I want to be able to use entire firebase implementation of main functions (`set`, `push`, `update`, `transaction`…). With this PR, I can work directly with firebase API docs and use any function easily.

To make short:
I provide all options of original firebase method each wrapper methods and add `transaction` method on FirebaseObjectObservable and FirebaseListObservable.
### Code sample

All samples are into docs & tests.

### CODE REVIEW

**Comment by katowulf** [comment]
Great work, @fabien0102! Thanks for following the PR template, including tests and docs, and generally being awesome. Hang in there while we find a few parsecs to review this.

**Comment by katowulf** [comment]
@jeffbcross @davideast what are your thoughts on this? It looks like we had consensus in #201 that we didn't want to wrap all the native methods. This seems to do exactly that.

I think what I'd rather see here are some short blurbs added to the docs explaining how to drop to the SDK level for these ops, perhaps with links to some appropriate core Firebase docs, rather than wrappers.

@fabien0102 my general objection to wrapping the API methods is that it creates a lot of hidden complexity and maintenance, and by extension bugs and rework--a subtlety we became all to familiar with in AngularFire 1.x. Unless we're adding value by wrapping the methods (i.e. taking away some boilerplate and work on the developer's part) this is unnecessary overhead. Since I don't see anything other than a straight wrapping of the core SDK methods, I see no reason to start down this path.

**Comment by vivanov-dp** [comment]
@katowulf I understand your argument and I'm +1, either way this is a case that needs a solution as it pops up in practice. I'm here after just having added .transaction() to FirebaseObjectObservable myself and I can tell I will need direct access to the firebase.database.Reference in future for .child() at least and probably for building queries, so next thing on my list is to add a getter for that and I know I will have to do some more reading trough the code to turn it back into a FirebaseObjectObservable.

**Comment by fabien0102** [comment]
@katowulf I must admit, I have see the consesus in #201, but I find ugly to must do `.$ref.transaction()` whereas I have a quick and proper access for all others firebase methods…

I could understand you argument of complexity if firebase have more of 4 methods for object/list ( transaction include). For my personal experience, I lose a lot of time to see how angularfire2 implements firebase methods (because of firebase doc api is actually more complete).

In short, I find angularfire2 very cool but I’m really not understand why the consensus is to implements 90% of firebase methods (missing optional args, even if I don’t need to use it, it’s cost nothing to pass them) and this `transaction` boycott.

**Comment by theunreal** [comment]
I totally agree with @fabien0102, using `.transaction()` is common and angularfire2 should cover it aswell. The attitude of keeping it simple makes Angularfire2 fully cover small apps only. Big and medium apps are must to use the firebase `.$ref` and I find it inconvenient to mix both in the same component and app.
